cmd/snap/cmd_run.go

```diff
@@ -423,7 +441,118 @@ func migrateXauthority(info *snap.Info) (string, error) {
 	return targetPath, nil
 }
 
-func runSnapConfine(info *snap.Info, securityTag, snapApp, command, hook string, args []string) error {
+func straceCmd() ([]string, error) {
+	current, err := user.Current()
+	if err != nil {
+		return nil, err
+	}
+	sudoPath, err := exec.LookPath("sudo")
+	if err != nil {
+		return nil, fmt.Errorf("cannot use strace without sudo: %s", err)
+	}
+
+	// try strace from the snap first, we use new syscalls like
+	// "_newselect" that are known to not work with the strace of e.g.
+	// ubuntu 14.04
+	var stracePath string
+	cand := filepath.Join(dirs.SnapMountDir, "strace-static", "current", "bin", "strace")
+	if osutil.FileExists(cand) {
+		stracePath = cand
+	}
+	if stracePath == "" {
+		stracePath, err = exec.LookPath("strace")
+		if err != nil {
+			return nil, fmt.Errorf("cannot find an installed strace, please try: `snap install strace-static`")
+		}
+	}
+
+	return []string{
+		sudoPath, "-E",
+		stracePath,
+		"-u", current.Username,
+		"-f",
+		"-e", "!select,pselect6,_newselect,clock_gettime",
+	}, nil
+}
+
+func runCmdUnderStrace(origCmd, env []string, opts runOptions) error {
+	// prepend strace magic
+	cmd, err := straceCmd()
+	if err != nil {
+		return err
+	}
+	// append strace options - TODO: use shlex
+	for _, opt := range strings.Split(opts.straceOpts, " ") {
+		if strings.TrimSpace(opt) != "" {
+			cmd = append(cmd, opt)
+		}
+	}
+	cmd = append(cmd, origCmd...)
+
+	// run with filter
+	gcmd := exec.Command(cmd[0], cmd[1:]...)
+	gcmd.Env = env
+	gcmd.Stdin = Stdin
+	gcmd.Stdout = Stdout
+	stderr, err := gcmd.StderrPipe()
+	if err != nil {
+		return err
+	}
+	filterDone := make(chan int)
+	var straceErrorOutput []string
```

Comment:
`straceErrorOutput` does not appear to be used

Reply:
Indeed, that is a merge artefact, I will remove it.